docs/tests.md

```diff
@@ -0,0 +1,454 @@
+against the KAT values.
+
+- Create a new file in the `tests/internal_estimators` directory with a name
+  that matches your KAT generator file from step 1, but without the `ext_`
+  prefix. For instance, if you created `tests/external_estimators/ext_sdfq.py`,
+  your new file would be `tests/internal_estimators/sdfq.py`.
+
+- Within this new file, define a corresponding internal estimation function for
+  each KAT generator function from step 1. Use the same function name but remove
+  the `ext_` prefix. For example, `ext_lee_brickell` would have a corresponding
+  `lee_brickell` function.
+
+- These internal estimation functions **must** adhere to the following
+  structure:
+
+  1. **Parameters:**
+
+     - `input`: A tuple containing the input parameters received from the
+       serialized KAT data.
+     - `epsilon`: A hardcoded value representing the acceptable error tolerance
+       during comparison. This helps account for potential small discrepancies
+       arising from different calculation methods or precision levels.
+
+  2. **Returns:**
+
+     - `actual_complexity`: The estimated complexity value as calculated by your
+       internal estimator or algorithm.
+     - `epsilon`: Returns the same `epsilon` value passed as input. This
+       practice maintains consistency and makes it clear that error tolerance is
+       being considered.
+
+For example, the `lee_brickell` function would look like this:
+
+```python
+def lee_brickell(input, epsilon = 0.01):
+    """Estimate produced by the CryptographicEstimators library for the Lee-Brickell SDFq problem.
+
+    This function calculates the complexity estimate for a single case of the Lee-Brickell SDFq problem.
+
+    Args:
+        input (Tuple[int, int, int, int]): A tuple containing (n, k, w, q) parameters for the SDFq problem.
+        epsilon (float): The maximum error tolerance for this estimation.
+
+    Returns:
+        Tuple[float, float]: A tuple containing:
+            - float: The actual complexity calculated by the LeeBrickell algorithm.
+            - float: The epsilon value used for error tolerance.
+    """
+    n, k, w, q = input  # Unpack input tuple
+
+    actual_complexity =  ...# Calculate the actual complexity using your library's implementation
+
+    return actual_complexity, epsilon  # Return the calculated complexity and the epsilon
+```
+
+#### 4. Run Your Tests
+
+With your KAT generator functions, internal estimator implementations, and
+generated reference values in place, you're ready to run your tests! Use the
+following command from your library's root directory:
+
+```bash
+make docker-pytest
+```
+
+Or by manually executing
+`pytest tests/validations/test_<your_new_estimator>.py`. The output will
+indicate whether your internal estimators align with the expected KAT values
+within the defined error tolerance.
+
+## Doctests (with pytests)
+
+TODO
+
+## Writing Doctests (with sage doctests)
```

Comment:
Should we remove this section? Are we waiting from the full transition of the docstring to happen? 
It seems to me since it is in a different PR it might be good to remove it now.

Reply:
Agree. Applied.